helix-view/src/document.rs

```diff
@@ -16,6 +16,7 @@ use std::collections::HashMap;
 #[derive(Copy, Clone, PartialEq, Eq, Hash)]
 pub enum Mode {
     Normal,
+    Space,
```

Comment:
Err, space become a mode on itself? Why do we want that? Isn't it just normal mode?

Reply:
https://github.com/helix-editor/helix/blob/master/helix-term/src/commands.rs#L2277 makes it easier to display it. I couldn't think of a non-hacky method for displaying the mode. It also does not include the normal mode keybindings, so it kinda makes sense to actually have it as a separate mode.

Reply:
^ this is why I made the change... Was testing out helix and typed space in normal mode and was confused why nothing was happening.

Reply:
So I had `space` and `goto` both as real modes initially, but I converted them to `on_next_key` when it was implemented -- it makes more sense since it's just a transient mode you enter for a single keypress then exit back to normal mode.

I think the way to go would be to take a name in the `cx.on_next_key` callback, then display that instead of doc.mode() in the statusbar if available. `on_next_key` will have to carry more data later on anyway since we also want to display a help popup for these transient modes.